tests/test_load_fund_data.py

```diff
@@ -0,0 +1,238 @@
+        "Config",
+        (),
+        {"database_path": "path/to/db.sqlite", "input_data_path": "path/to/data"},
+    )
+    database_path, input_data_path, fund_holdings_table_name = get_config_n_constant()
+    assert database_path == "path/to/db.sqlite"
+    assert input_data_path == "path/to/data"
+    assert fund_holdings_table_name == "mocked_fund_holdings"
+
+
+@pytest.mark.parametrize(
+    "fund_name_raw, expected_cleaned_name",
+    [
+        ("TT_monthly_Trustmind", "Trustmind"),
+        ("rpt-Catalysm", "Catalysm"),
+        ("Fund Whitestone", "Whitestone"),
+        ("Report-of-Gohen", "Gohen"),
+        ("mend-report Wallington", "Wallington"),
+        ("Applebead", "Applebead"),
+        (" ", None),
+        ("!@#$%^&*()", None),
+    ],
+)
+def test_clean_fund_name(fund_name_raw, expected_cleaned_name):
+    """Test clean_fund_name function with various fund name formats."""
+    cleaned_name = clean_fund_name(fund_name_raw)
+    assert cleaned_name == expected_cleaned_name
+
+
+@pytest.mark.parametrize(
+    "date_raw, expected_cleaned_date",
+    [
+        ("12-05-2025", "2025-05-12"),
+        ("05-12-2025", "2025-12-05"),
+        ("12_05_2025", "2025-05-12"),
+        ("2025-05-12", "2025-05-12"),
+        ("20250512", "2025-05-12"),
+        ("12/05/2025", None),
+        ("05/12/2025", None),
+        ("2025/05/12", None),
+        ("invalid_date", None),
+        (" ", None),
+        ("!@#$%^&*()", None),
+    ],
+)
+def test_clean_date(date_raw, expected_cleaned_date):
+    """Test clean_date function with various date formats."""
+    cleaned_date = clean_date(date_raw)
+    assert cleaned_date == expected_cleaned_date
+
+
+@patch("load_fund_data.clean_fund_name")
+@patch("load_fund_data.clean_date")
+def test_parse_filename_valid(mock_clean_date, mock_clean_fund_name):
+    """Test parse_filename function with valid filename."""
+    mock_clean_fund_name.return_value = "FundName"
+    mock_clean_date.return_value = "2025-05-12"
+
+    filename = "FundName.20250512.csv"
+    fund_name, date = parse_filename(filename)
+
+    assert fund_name == "FundName"
+    assert date == "2025-05-12"
+
+
+@patch("load_fund_data.clean_fund_name")
+@patch("load_fund_data.clean_date")
+def test_parse_filename_invalid(mock_clean_date, mock_clean_fund_name):
+    """Test parse_filename function with invalid fund name and date."""
+    mock_clean_fund_name.return_value = None
+    mock_clean_date.return_value = None
+    with pytest.raises(ValueError):
+        parse_filename("!@#.InvalidDate.csv")
+
+
+def test_parse_filename_invalid_format_assertion():
+    """Test parse_filename function with invalid filename format not consistent with fund.date.csv."""
+    with pytest.raises(AssertionError, match="Invalid filename format"):
+        parse_filename("InvalidFilename.csv")
+
+
+def test_pandera_to_polars_dtype():
+    """Test pandera_to_polars_dtype conversion."""
+    assert pandera_to_polars_dtype(pa.String()) == pl.String
```

Comment:
Now this function is perhaps too simple to test! It's essentially a dict lookup.